internal/pkg/cri/pods.go

```diff
@@ -104,21 +106,35 @@ func (c *Client) StopAndRemovePodSandboxes(ctx context.Context, stopAction StopA
 	for _, pod := range pods {
 		pod := pod // https://golang.org/doc/faq#closures_and_goroutines
 
-		status, _, err := c.PodSandboxStatus(ctx, pod.GetId())
-		if err != nil {
-			return err
-		}
+		g.Go(func() error {
+			status, _, e := c.PodSandboxStatus(ctx, pod.GetId())
+			if e != nil {
+				if grpcstatus.Code(e) == codes.NotFound {
+					return nil
+				}
 
-		networkMode := status.GetLinux().GetNamespaces().GetOptions().GetNetwork()
+				return e
```

Comment:
moved this into the `errgroup`, as otherwise `return err` leaves some goroutines in `errgroup` running while we return